src/Damage/ProbabilisticDamagePolicy.cc

```diff
@@ -0,0 +1,346 @@
+
+              // Find how many flaws are activated in this direction
+              CHECK(maxFlaw(i) > 0.0);
+              double fractionFlawsActive = 1.0;
+              if (numFlaws(i) > 1 and strainj < maxFlaw(i)) {
+                CHECK(maxFlaw(i) > minFlaw(i));
+                const auto Nmini = pow(minFlaw(i), mmWeibull);  // Missing factors of k*Vi, but those cancel below
+                fractionFlawsActive = ((pow(strainj, mmWeibull) - Nmini)/
+                                       (pow(maxFlaw(i), mmWeibull) - Nmini));
+              }
+              CHECK(fractionFlawsActive >= 0.0 and fractionFlawsActive <= 1.0);
+
+              // Choose the allowed range of D.
+              double Dmin, Dmax;
+              if (multiplier >= 0.0) {
+                Dmin = D0;
+                Dmax = std::max(D0, fractionFlawsActive);
+              } else {
+                Dmin = 0.0;
+                Dmax = D0;
+              }
+              CHECK(Dmax >= Dmin);
+              CHECK(Dmin >= 0.0 && Dmax <= 1.0);
+
+              // Increment the damage.
+              const auto D013 = FastMath::CubeRootHalley2(D0);
+              const auto D113 = D013 + multiplier*fractionFlawsActive*DDDt(i);
+              const auto D1 = max(Dmin, min(Dmax, FastMath::cube(D113)));
+              CHECK((D1 - D0)*multiplier >= 0.0);
+
+              // Increment the damage tensor.
+              Di += (D1 - D0)*strainDirection.selfdyad();
+            }
+          }
+        }
+
+        // Enforce bounds on the damage.
+        const auto Dvals = Di.eigenValues();
+        if (Dvals.minElement() < 0.0 or
+            Dvals.maxElement() > 1.0) {
+          const auto Deigen = Di.eigenVectors();
+          Di = constructSymTensorWithBoundedDiagonal(Deigen.eigenValues,
+                                                     1.0e-5,
+                                                     1.0);
+          Di.rotationalTransform(Deigen.eigenVectors);
+        }
+        ENSURE(Di.eigenValues().minElement() >= 0.0 and
+               fuzzyLessThanOrEqual(Di.eigenValues().maxElement(), 1.0, 1.0e-5));
+        //     ENSURE(fuzzyGreaterThanOrEqual(Di.eigenValues().minElement(), 0.0, 1.0e-5) &&
+        //            fuzzyLessThanOrEqual(Di.eigenValues().maxElement(), 1.0, 1.0e-5));
+        //     ENSURE(fuzzyGreaterThanOrEqual(Di.eigenValues().minElement(), 0.0) &&
+        //            fuzzyLessThanOrEqual(Di.Trace(), 1.0));
+      }
+    }
+  }
+}
+
+//------------------------------------------------------------------------------
+// Equivalence operator.
+//------------------------------------------------------------------------------
+template<typename Dimension>
+bool
+ProbabilisticDamagePolicy<Dimension>::
+operator==(const UpdatePolicyBase<Dimension>& rhs) const {
+
+  // We're only equal if the other guy is also an increment operator.
+  const ProbabilisticDamagePolicy<Dimension>* rhsPtr = dynamic_cast<const ProbabilisticDamagePolicy<Dimension>*>(&rhs);
```

Comment:
It look like this just performing an equivalence operation on type compatibility correct? Do we have type traits for Policies like these to make these kinds of comparisons, not a hard PR change requirement but if we make these types of comparisons for other policies that could be something to think about adding later on

Reply:
That's common to many of these policices.  Might be worth a focused discussion at some point.